Core/Core/Models/CommitObjectBuilder.cs

```diff
@@ -105,7 +105,7 @@ protected void ApplyRelationships(IEnumerable<Base> toAdd, Base rootCommitObject
       {
         ApplyRelationship(c, rootCommitObject);
       }
-      catch (Exception ex)
+      catch (Exception ex) when (!ex.IsFatal())
```

Comment:
Feel free to resolve but I gotta wonder when I see these sorts of things, but what if this gets caught, to the caller ApplyRelationships() worked :O

Reply:
I think this was to avoid some edge cases where we've setup relationships for objects that are invalid, and we wanted to ignore those objects.
But perhaps there's a cleaner way we can check & throw for this.